mobile/lib/src/presentation/pages/settings/settings_page.dart

```diff
@@ -0,0 +1,186 @@
+import 'package:polydodo/src/domain/settings/settings.dart';
+import 'package:polydodo/src/presentation/navigation/navdrawer_widget.dart';
+import 'package:polydodo/src/presentation/widgets/loading_indicator.dart';
+import 'package:settings_ui/settings_ui.dart';
+
+class SettingsPage extends StatefulWidget {
+  SettingsPage({Key key}) : super(key: key);
+
+  @override
+  _SettingsPageState createState() => _SettingsPageState();
+}
+
+class _SettingsPageState extends State<SettingsPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text('Settings')),
+      drawer: NavDrawer(activeTab: NavdrawerTab.Settings),
+      body: BlocBuilder<SettingsCubit, SettingsState>(
+        builder: (context, state) {
+          return state is SettingsLoadSuccess
+              ? SettingsList(
+                  sections: [
+                    SettingsSection(
+                      title: 'Personnal informations',
+                      tiles: [
+                        SettingsTile(
+                          title: 'Age',
+                          subtitle: 'In years',
+                          leading: Icon(Icons.cake),
+                          trailing: TextButton(
+                              child: Text(state.settings.age == null
+                                  ? 'Empty'
+                                  : state.settings.age.toString()),
+                              onPressed: () => _showDatePicker(context)),
+                        ),
+                        SettingsTile(
+                          title: 'Sex',
+                          subtitle: 'Your biological sex',
+                          leading: Icon(Icons.face),
+                          trailing: _SexButton(
+                            savedSex: state.settings.sex,
+                            onSelected: (newSex) =>
+                                BlocProvider.of<SettingsCubit>(context)
+                                    .setSex(newSex),
+                            activeStyle: TextStyle(fontWeight: FontWeight.bold),
+                            defaultStyle: TextStyle(),
+                          ),
+                        ),
+                        SettingsTile(
+                          title: 'Acquisition Board',
+                          subtitle: 'Your OpenBCI board',
+                          leading: Icon(Icons.memory),
+                          trailing: _BoardButton(
+                            savedBoard: state.settings.board,
+                            onSelected: (newBoard) =>
+                                BlocProvider.of<SettingsCubit>(context)
+                                    .setBoard(newBoard),
+                            activeStyle: TextStyle(fontWeight: FontWeight.bold),
+                            defaultStyle: TextStyle(),
+                          ),
+                        ),
+                      ],
+                    ),
+                  ],
+                )
```

Comment:
This is too repetitive. I recommend creating a factory method for SettingsTile.

Reply:
I would also extract the whole SettingsList into a settings_list_section.dart file and have a buildSettingsList() function to make it easier to read.

Reply:
Because I added some genericity (?), it now feels a lot less repetitive. I'm open if you have more suggestions or if the old one still stands!

Reply:
@abelfodil @MouradLachhab  J'vais vous avouer c'est pt'être un peu ghetto haha. Mais c'est les limitations de dart avec les enums mixed avec les types génériques.

J'pense que c'est quand même clean au final. Et ça limite la répétition.